### ISSUE TITLE
Uncaught Error: Malformed UTF-8 data error

### ISSUE THREAD

**Issue by joecwu** [blcksync/bc-ipfs#26]
I got `Uncaught Error: Malformed UTF-8 data` when I tried to accessing file.
here is the hash Id I used `QmXWjKeiHHm6Pokb6WitX2BvbeqM5x3tkzKBMKmZEheC6E`

```
App.js:389 Capturing input from access_ipfs_metadata with value = QmXWjKeiHHm6Pokb6WitX2BvbeqM5x3tkzKBMKmZEheC6E
01:55:12.446 App.js:407 Accessing with metadata = QmXWjKeiHHm6Pokb6WitX2BvbeqM5x3tkzKBMKmZEheC6E and encryptedTxt = 
01:55:12.446 App.js:411 Identified contract address = 0x16C60A50c0d9E2C191370E42aA9d2FB22B99F1fB
01:55:12.458 App.js:417 All available accounts: 0xD4F84cB84c024235263702dE37982a0C0eFB666e
01:55:12.459 App.js:419 Applying the first eth account[0]: 0xD4F84cB84c024235263702dE37982a0C0eFB666e for contract 0x16C60A50c0d9E2C191370E42aA9d2FB22B99F1fB
01:55:17.608 App.js:430 decryptIPFS tx=0xbd0b53f45e522090c8e533a770bc1c7388232dbe353030ed213555e1b7f0a3c6
01:55:43.562 App.js:448 fetching decrypted 1st_partial_key=dRomORNXQcpNF8Hc82dE5c8a7H354pHAPhYJ7oZGumChCyN2fddR2rDWb4pjwdVATuLAyhyS3Vb9S6IWw9r9khLhhdcVfxPIYo3QxqJ8WnZfjaHx9yVvtYyO7qtCEXACayjMTVOsmv3G9nrPIcGpzJRaFps3dNt4AjH8G9UEXSAh2Hi7vDzmqnCyaVs6pnsyv2SbmXCpRbLMGSjSBMmOSTTTIGJUZC1v4y3qdMKWzXxuHkaHlfDx6veltwHV3aek4LEMQEiTFBlHFjVYebp3d3sDnrMuX30fKFBnizjFJ62LSKdUNYtpzxeXwwLc8zqIaMwBOXnmheON5GhDXu0p5IHUgayWrVsZzASdAOstR8UDPFTcJzJfM9cHrICo6KlzYaiwftlz33mikMhdMoJ855KSMKzXzAFyhKgt5oWKurAlbrMxRs3wqWjQsghUTaWXu2APtA7PqlOqbzzVjk66oy3dQXJd8nv1ePMHvJxfTnSxM4JgI1S1ubeNXNG8dlOGJFZOQok8TjYUnaKze9Wot6wDrIk7sFJgw3BLO0aafpprmEM1Ms3NP6JaYTMcmgMiMERa31iRgQZPNh8DI0CQfQuWFb4THv7Xwto5JtuwLu37PLSKduZ1mMtBR2eOQS1TvvEqmJ8UjOTri03drP1qOzEZjgCc354zNpRZoVobhLTJ0rjQFD25GL7yF8LW6mhJQmd4XHJ4qpYRGNZwtNoNLk49FSz7Vm1NcRtHwUczScBLimoVYAlpIT7kVWJE8Fu6yqZL3QMWsobhdTvC1v4PBu9YRgzgmfGsXDa6DO9WbmG34y1k1GXsoRSeTKkUDhVhnOOERVgpJGn5E0CkiP8WfCGRyjS6pY6KXPDTsrsozfnXrjBupIPvTXYL1sIB9yJPO4uAB6X9WYUayGC8I4p6SMgjTcQ65je2A34Avyo5ovUsx8PueN2dUaF2saje3ksMbCmE 2nd_partial_key=37 encryptedHash=U2FsdGVkX1+++/ewlvWjJ5sGgoYcLYzXGrPK/t8pB1B/KgJMryVbQKpWUiEQcS2fGQv5VJ/19TG6hf0/HQDeqg== cost=4181745000000
01:55:43.568 core.js:478 Uncaught Error: Malformed UTF-8 data
    at Object.stringify (core.js:478)
    at WordArray.init.toString (core.js:215)
    at Object.callback (App.js:459)
    at sendTxCallback (index.js:484)
    at index.js:147
    at inpage.js:1
    at inpage.js:1
    at i (inpage.js:1)
    at inpage.js:1
    at inpage.js:1
```

**Comment by matr1xc0in** [comment]
Note: After discussing with @joecwu , I have try to reproduce this and I couldn't. It could be some local cache errors or something else.

**Comment by matr1xc0in** [comment]
It turns out we are still having issues using CryptoJS. I would vote that we switch to `aes-js` https://www.npmjs.com/package/aes-js

**Comment by matr1xc0in** [comment]
It turns out it could be due to https://github.com/blcksync/bc-ipfs/issues/72 as a side effect. If the file was registered with empty hash, during access, it will also cause the IPFS hash to be empty. We will continue to monitor this since we have fixed https://github.com/blcksync/bc-ipfs/issues/72 with PR https://github.com/blcksync/bc-ipfs/pull/73

**Comment by al-blockmed** [comment]
This is fixed in PR72 and PR73. The root cause was coming from an empty IPFS hash, so of course, it will be an empty string here. This was caused from upstream code passing empty IPFS hash to downstream code.

**Comment by matr1xc0in** [comment]
This is fixed in #72 and #73. The root cause was coming from an empty IPFS hash, so of course, it will be an empty string here. This was caused from upstream code passing empty IPFS hash to downstream code.
